tests/31sync/00user.pl

```diff
@@ -0,0 +1,10 @@
+push our @EXPORT, qw( matrix_register_sync_user );
+
+my $counter = 0;
+
+sub matrix_register_sync_user {
```

Comment:
Any particular reason for this new function? The regular `matrix_register_user` now generates serial-numbered user IDs if not supplied; so the only thing you're really saving here is the `with_events => 0`.